lib/node_modules/@stdlib/array/base/cuany-by/lib/assign.js

```diff
@@ -0,0 +1,322 @@
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} predicate - test function
+* @param {*} [thisArg] - execution context
+* @returns {Collection} output array
+*
+* @example
+* function isPositive( v ) {
+*     return v > 0;
+* }
+*
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = toAccessorArray( [ false, false, false, true, false ] );
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, false, null ] );
+*
+* var arr = accessors( arraylike2object( x ), arraylike2object( y ), 2, 0, isPositive );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns false
+*
+* v = y.get( 4 );
+* // returns false
+*
+* v = y.get( 6 );
+* // returns true
+*
+* v = y.get( 8 );
+* // returns true
+*/
+function accessors( x, y, stride, offset, predicate, thisArg ) {
+	var xdata;
+	var ydata;
+	var xget;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	xdata = x.data;
+	ydata = y.data;
+
+	xget = x.accessors[ 0 ];
+	yset = y.accessors[ 1 ];
+
+	flg = false;
+	io = offset;
+	for ( i = 0; i < xdata.length; i++ ) {
+		if ( flg === false && predicate.call( thisArg, xget( xdata, i ), i, x) ) {
+			flg = true;
+		}
+		yset( ydata, io, flg );
+		io += stride;
+	}
+	return ydata;
+}
+
+/**
+* Cumulatively tests whether at least one element in a provided array passes a test implemented by a predicate function.
+*
+* @private
+* @param {Collection} x - array containing interleaved real and imaginary components
+* @param {Object} y - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} predicate - test function
+* @param {*} [thisArg] - execution context
+* @returns {Collection} output array
+*
+* @example
+* function isPositive( v ) {
+*     return v > 0;
+* }
+*
+* var Float64Array = require( '@stdlib/array/float64' );
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Float64Array( [ 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ] );
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, false, null ] );
+*
+* var arr = complex( x, arraylike2object( y ), 2, 0, isPositive );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns true
+*
+* v = y.get( 4 );
+* // returns true
+*
+* v = y.get( 6 );
+* // returns true
+*
+* v = y.get( 8 );
+* // returns true
+*/
+function complex( x, y, stride, offset, predicate, thisArg ) {
+	var ydata;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	yset = y.accessors[ 1 ];
+	ydata = y.data;
+
+	flg = false;
+	io = offset;
+	for ( i = 0; i < x.length; i += 2 ) {
+		if ( flg === false && ( predicate.call( thisArg, x[ i ], i, x ) || predicate.call( thisArg, x[ i+1 ], i, x ) ) ) {
```

Comment:
@kgryte Ergonomics when using `assign` here with a complex array seem not ideal to me, as users won't know whether a `predicate` function is getting called with the real or imaginary component. Thoughts?